lsp-server/src/test/scala/org/alephium/ralph/lsp/server/RalphLangServerSpec.scala

```diff
@@ -104,80 +104,62 @@ class RalphLangServerSpec extends AnyWordSpec with Matchers with MockFactory wit
 
       server.exitWithCode() shouldBe 0
       server.getState().listener.get.isCancelled shouldBe true
-   }
+    }
 
     "exit with error if shutdown wasn't requested" in {
       val server = RalphLangServer(client, listener)
 
       server.exitWithCode() shouldBe 1
       server.getState().listener.get.isCancelled shouldBe true
-   }
+    }
   }
 
   "registerClientCapabilities" should {
+    "register watched files if dynamic registration is true" in {
+      implicit val compiler: CompilerAccess = null // compile is never accessed
+      implicit val file: FileAccess = null // file/disk IO is never accessed
 
-    def buildServer(dynamicRegistration: Option[Boolean], registered: AtomicBoolean) = {
-      implicit val compiler: CompilerAccess = CompilerAccess.ralphc
-      implicit val file: FileAccess = FileAccess.disk
-      val client: RalphLangClient = new EmptyRalphLangClient {
-        override def registerCapability(params: RegistrationParams): CompletableFuture[Void] = {
-          registered.set(true)
-          new CompletableFuture[Void]()
-        }
-      }
-      val listener = CompletableFuture.runAsync(() => ())
-      val server = RalphLangServer(client, listener)
-
-      val capabilities = dynamicRegistration match {
-        case None => new ClientCapabilities()
-        case Some(dynamicRegistration) =>
-          val watchedFilesCapabilities = new DidChangeWatchedFilesCapabilities(dynamicRegistration)
-          val workspaceCapabilities = new WorkspaceClientCapabilities()
-          workspaceCapabilities.setDidChangeWatchedFiles(watchedFilesCapabilities)
-          val result = new ClientCapabilities()
-          result.setWorkspace(workspaceCapabilities)
-          result
-      }
+      val client = mock[RalphLangClient]
 
-      val initialise = new InitializeParams()
-      val workspaceURI = new URI("file://test")
-      initialise.setRootUri(workspaceURI.toString)
-      initialise.setCapabilities(capabilities)
+      val server =
+        RalphLangServer(
+          client = client,
+          listener = CompletableFuture.runAsync(() => ()),
+          clientAllowsWatchedFilesDynamicRegistration = true
+        )
 
-      server.initialize(initialise).get()
-      server
-    }
+      server.getState().clientAllowsWatchedFilesDynamicRegistration shouldBe true
 
-    "register watched files if dynamic registration is true" in {
-      val registered = new AtomicBoolean(false)
-      val server = buildServer(Some(true), registered)
-      server.registerClientCapabilities()
+      // register is called once with the default client-capabilities
+      (client.register _)
+        .expects(RalphLangServer.clientCapabilities())
+        .returning(new CompletableFuture[Void]())
+        .once()
```

Comment:
damn you're such a pro, I can never make my mock to work

Reply:
[This flag](https://github.com/alephium/ralph-lsp/pull/78/commits/9caaedb7d39ca51b5243c27d706c8eb0a171bf11) that was previously missing might have been the cause of the mock not working. Hopefully we won't need this flag in a future Scala release.

Reply:
:astonished: